src/utilities/toPascalCase.ts

```diff
@@ -0,0 +1,26 @@
+import {upperCaseFirstCharacter} from './upperCaseFirstCharacter'
+
+export const toPascalCase = (string: string | string[]) => {
+  if (!Array.isArray(string)) {
+    string = [string]
+  }
+  // match unsupported characters
+  const regex = /[^a-zA-Z0-9]+/g
+  // replace any non-letter and non-number character and split into word array
+  const stringArray = string
+    .filter(part => part !== '@')
+    .filter(Boolean)
+    .join(' ')
+    .replace(regex, ' ')
+    .split(' ')
+  return (
+    stringArray
+      // remove undefined if exists
+      .filter((part: unknown): part is string => typeof part === 'string')
+      // ucFirst all but first part
+      .map((part: string) => {
+        return upperCaseFirstCharacter(part)
+      })
+      .join('')
```

Comment:
Another nit (not blocking) but you could do all array manipulation inside the function body and just return the final output in the return to make this a bit clearer. Right now it seems like there's the same things happening in both the return and function body.

Reply:
I thought about this as well. Will look into it. 👍